api/pkg/apis/v1alpha1/providers/states/k8s/k8s.go

```diff
@@ -189,8 +189,12 @@ func (s *K8sStateProvider) Upsert(ctx context.Context, entry states.UpsertReques
 		Resource: resource,
 	}
 
-	j, _ := json.Marshal(entry.Value.Body)
+	if entry.Value.ID == "" {
+		err := v1alpha2.NewCOAError(nil, "found invalid request ID", v1alpha2.InternalError)
```

Comment:
It's better to return BadRequest instead of InternalError?

Reply:
Updated.